### ISSUE TITLE
MSID: 87720 Version: 2 DOI: 2023.04.04.535526

### ISSUE THREAD

**Issue by nlisgo** [elifesciences/enhanced-preprints-import#856]
[PLACE MANUSCRIPT AND EDITOR DETAILS HERE WHEN AVAILABLE] See step 3

[PLACE PDF URL HERE WHEN AVAILABLE] See step 7

## Step 1. Inform bioRxiv

Who can help: @QueenKraken, @nlisgo, @scottaubrey

- [ ] Manuscript is in data hub index (https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v2/by-publisher/elife/get-by-manuscript-id?manuscript_id=87720)

or (only one should be ticked. remove other from description.)

- [ ] Confirmed with Ted at bioRxiv that MECA archive is available

Send the following email to Ted and wait for his reply.

```
Hi Ted,

Please can you prepare the preprint MECA for 10.1101/2023.04.04.535526

Thanks
```

## Step 2. Create preview of manuscript

Who can help: @fred-atherden, @nlisgo, @scottaubrey

- [ ] Manuscript is available for preview (https://prod--epp.elifesciences.org/preview/87720/v2)
    - [ ] Trigger import for staging: https://prod--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://prod-elife-epp-data/data
    - [ ] Trigger import for staging: https://staging--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://staging-elife-epp-data/data
    - [ ] Pull request created to [enhanced-preprint-data](https://github.com/elifesciences/enhanced-preprints-data/pulls)
    - [ ] Post link to pull request below

Pull request: [PLACE LINK TO PULL REQUEST HERE]

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-manuscript

## Step 3: Awaiting public reviews and QC

Who can help: Production team

- [ ] QC passed or QC not needed

Who can help: Editorial team

- [x] Public reviews available on sciety (https://sciety.org/articles/activity/10.1101/2023.04.04.535526)
- [ ] Manuscript and editor details at the top of this issue (Supplied by Editorial team)
- [ ] Reviews returned by EPP - https://prod--epp.elifesciences.org/api/reviewed-preprints/87720/v2/reviews

<details>
<summary>Example</summary>

```
"msas": "Genetics and Genomics", "Neuroscience"
"msid": "84628"
"version": "1"
"preprintDoi": "10.1101/2022.10.28.514241"
"articleType": "Reviewed Preprint"
"status": "Published from the original preprint after peer review and assessment by eLife."

"Reviewed Preprint posted": "2023-01-02"
"Sent for peer review": "2022-10-28"
"Posted to bioRxiv": "2022-11-21" (link: "Go to bioRxiv": "https://www.biorxiv.org/content/10.1101/2022.10.28.514241v1")

Editors:
Reviewing Editor
Michael B Eisen
University of California, Berkeley, United States
Senior Editor
Michael B Eisen
University of California, Berkeley, United States
```

</details>

## Step 4: Modify manuscripts.json (no PDF)

- [ ] Reviewed preprint is published to EPP (https://elifesciences.org/reviewed-preprints/87720)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below
- [ ] Request that a doi be registered
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

Pull request: [PLACE LINK TO PULL REQUEST HERE]
[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

<details>
<summary>Instructions to modify manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/publish-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/334/files

Once the pull request is merged in it should be available a few minutes later.

</details>

### Request that a doi

Post the following in [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77):

```
@Fred can you register a doi for https://elifesciences.org/reviewed-preprints/87720
```

## Step 5: Awaiting search reindex

- [ ] Reviewed preprint is avaliable on journal homepage (https://elifesciences.org)

The search reindex is triggered once an hour. We need the reviewed preprint to be indexed as the search application serves the journal homepage.

<details>
<summary>Additional info</summary>

If needed, the jenkins pipeline to reindex search can be triggered sooner.

https://alfred.elifesciences.org/job/process/job/process-reindex-reviewed-preprints/

</details>

## Step 6: Published! Request PDF generation

- [ ] Post the link to the PDF url at the top of the issue
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

Post the following to the [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) on slack:

```
@Ryan Dix-Peek please can you generate a PDF for https://elifesciences.org/reviewed-preprints/87720
```

## Step 7: Introduce PDF to data folder and git repo

- [ ] PDF is avaliable at https://github.com/elifesciences/enhanced-preprints-data/raw/master/data/87720/v2/87720-v2.pdf

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-pdf

## Step 8: Add PDF url to manuscripts.json

- [ ] Reviewed preprint PDF is available for download (https://prod--epp.elifesciences.org/reviewed-preprints/87720)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below

[PLACE LINK TO PULL REQUEST HERE]

<details>
<summary>Instructions to add PDF url to manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/add-pdf-url-to-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/397/files

Once the pull request is merged in it should be available a few minutes later.

</details>

## Step 9: Done!

- [ ] Kettle is on!

**Comment by naushinthomson** [comment]
Section 9.11 is missing in EPP (including supplementary figure 15)

![Image](https://github.com/elifesciences/enhanced-preprints-import/assets/57144084/b52f4893-1bd9-4e6c-9317-59103e3555c1)
![Image](https://github.com/elifesciences/enhanced-preprints-import/assets/57144084/afc57ced-6cb1-4408-bde6-d513637e5788)

**Comment by fred-atherden** [comment]
Thanks for these @naushinthomson - great spots.

For [the issue with section 9.11](https://github.com/elifesciences/enhanced-preprints-import/issues/856#issuecomment-1680159084), I think this is a V2 vs. V3 issue. Since the authors have submitted for an RPv3 and their [V3 revision summary on bioRxiv](https://www.biorxiv.org/content/10.1101/2023.04.04.535526v3.article-info) suggests the V3 is for that one ("`The manuscript has been revised after a second round of peer review at eLife.`"), I think we should be using preprint V2 for the RPV2. 

For [the other issue](https://github.com/elifesciences/enhanced-preprints-import/issues/856#issuecomment-1680148865) to do with the junk characters, I notice that the authors have included these junk characters in [their V2 PDF](https://www.biorxiv.org/content/10.1101/2023.04.04.535526v2.full.pdf). 

@acollings Do you know if authors able to ask bioRxiv to correct previous versions (in this case ask biorxiv to re-typeset using a corrected PDF)?

If not, at a push we could get Exeter to typeset a package for the V2 if the authors provide a PDF without junk characters in it.

**Comment by acollings** [comment]
Yes please, we should use V2 of the preprint for RP V2. (There will be an RP V3 using preprint V3.)

I'll ask the authors about the junk characters.

**Comment by fred-atherden** [comment]
The authors have provided an updated word/PDF file. Using this we have a corrected MECA package to use - `s3://prod-elife-epp-meca/10.1101/2023.04.04.535526-v2-meca.zip`. 

I plan to upload this in [this PR](https://github.com/elifesciences/enhanced-preprints-data/pull/150) later today.

----------------------------


@QueenKraken, @nlisgo, @scottaubrey, @hazalCiplak  I think this is a slightly new, unforeseen, requirement that will affect automation. Here we have a bioRxiv preprint that needs a non-bioRxiv MECA package (or rather not the default one) for version 2 of the reviewed preprint (because this version of the preprint is full of junk characters and bioRxiv will not correct previous versions of preprints - this one has subsequent preprint versions which are updated/revised). 

I assume we'll need a way to manually update the `computer-file` `url` path in the docMap that that data hub exposes and EPP will rely on under automation. 

Let me know if/when this requires further discussion.

**Comment by QueenKraken** [comment]
@hazalciplak this feels like something that can be supported by updating the spreadsheet? Let me know if this is incorrect.

**Comment by HazalCiplak** [comment]
Hello,

Thank you for bringing this to my attention. Currently, we aren't receiving MECA paths from a spreadsheet, so we need to retrieve this data from BioRxiv's MECA path information/API. Ideally, the best solution is to update the API response directly from the BioRxiv site. Handling data issues at the source is generally more efficient (if it is possible).

However, if updating the source directly is not possible, I can implement a specific process in the DocMaps API code for this particular manuscript and its related version. Please note that this might make the code less organized and more of a patchwork solution. I can proceed with this approach to unblock the situation, but if feasible, I recommend addressing the issue at the source.

For the confirmation, do you want to see something like below?

<img width="803" alt="Screenshot 2023-08-22 at 15 53 40" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/83723855-e033-4f82-88e1-eade6f0336f6">

cc: @de-code

**Comment by fred-atherden** [comment]
Thanks Hazal. I agree - it's definitely preferable to use the source data from bioRxiv.

Unfortunately if they have a policy against updating/correcting previous versions of a preprint it may hamper us though. Hopefully this will be a rare occurrence, but it might be something we come up against again in the future. If possible (and if what you mention above is not already), it may be worth considering an approach that can be extended to other preprints/versions as well.


The meca path for the version 2 looks good in your screenshot above, that would be ideal - thanks. 

It may be unrelated (or just test data that can be ignored), but I'm not sure that the version 1 meca path is correct. While that version _is_ taken from the bioRxiv API, the meca we have loaded for version 1 of this reviewed preprint is `s3://transfers-elife/biorxiv_Current_Content/April_2023/05_Apr_23_Batch_1572/1995c912-6c01-1014-b54a-f0c0b033df53.meca`.

**Comment by fred-atherden** [comment]
Now that `s3://prod-elife-epp-meca/10.1101/2023.04.04.535526-v2-meca.zip` is loaded, the issues mentioned above are now fixed.

**Comment by HazalCiplak** [comment]
> Now that `s3://prod-elife-epp-meca/10.1101/2023.04.04.535526-v2-meca.zip` is loaded, the issues mentioned above are now fixed.

Ah great! Thank you for letting us know. Do you mind me asking how this has been solved? Additionally, is there still a need to explore potential solutions for similar cases in the future?

**Comment by fred-atherden** [comment]
Ah sorry for the confusion @HazalCiplak - I was referring to the issues mentioned in [this comment](https://github.com/elifesciences/enhanced-preprints-import/issues/856#issuecomment-1680148865) above, not the issue that I looped you and the EPP team in on.

The reason that's relevant is that it now unblocks publication of this reviewed preprint under the current (not automated) process (under the current process I am loading content to [this repo](https://github.com/elifesciences/enhanced-preprints-data), and then syncing the data in that repo with an s3 bucket and importing the data in that bucket into EPP. Under automation my understanding is that the MECA path will be derived from the docmaps that data hub provides, but that isn't the case for the current process). 

We'll still need to explore the solution you mentioned above for the purposes of using the right MECA file under automation.

**Comment by fred-atherden** [comment]
Query from the authors around the terminology/wording of the assessment. Adding as blocked for now until this is resolved.

**Comment by acollings** [comment]
Assessment tweaked. I think it's fine to publish when we're able to. Thanks @fred-atherden! (Feel free to re-add the blocked label if we need it to indicate that we're waiting for Exeter XML etc!)

**Comment by HazalCiplak** [comment]
Hello,
I've prepared a [new Google Spreadsheet document](https://docs.google.com/spreadsheets/d/1mctCQuNFBjSn97Lihy7_vBO6z7-N-oqyLv4clyi6zHg/edit#gid=0) to facilitate the process of updating or overwriting the existing MECA paths. We can utilise this document for handling similar cases in the future. I've granted you edit permissions for the document. 

[The DocMap](https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v2/by-publisher/elife/get-by-manuscript-id?manuscript_id=87720) has already been updated accordingly. 

However, there is an issue with the preprint version match. The identifier `eLife-RP-RA-2023-87720R1` doesn't align with preprint version `2`; it corresponds to version `3`. This mismatch is occurring due to the hypothesis data. Plese find the related screenshot below:

<img width="863" alt="Screenshot 2023-08-24 at 08 23 42" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/ecb6b32c-1012-4c24-9960-276487f64b2a">

**Comment by fred-atherden** [comment]
Hi @HazalCiplak - this looks ideal, many thanks. I just have a few of questions:

1. Just so that I understand, in the spreadsheet should I use the `long_manuscript_identifier` column to target a particular version. So, for the sake of argument, if I wanted to update the version 1 of this article instead of version 2 I would add `eLife-RP-RA-2023-87720` instead of `eLife-RP-RA-2023-87720R1`? And likewise add `eLife-RP-RA-2023-87720R2` to change the path for version 3 of the reviewed preprint?

2. How often is the data from the spreadsheet consumed by datahub? Once a day at a certain time? How long should I expect to wait until the DocMap gets updated after adding to the spreadsheet?

3. Probably a question more for @scottaubrey and/or @nlisgo (and maybe something that's still to be determined), but under automation how do we trigger a reimport of this data after the DocMap has been updated with a different MECA path? 

@acollings regarding Hazal's point about the hypothesis data - is that something you need to fix? My understanding is that we're using v2 of the preprint for v2 of the reviewed preprint (but LMK if I've misunderstood).

**Comment by HazalCiplak** [comment]
Hi @fred-atherden,

These are very good points!

1- I've used `long_manuscript_identifier` since I couldn't locate the specific the preprint DOI and version (v2) in DocMaps. However, we can revert to using the preprint DOI and its corresponding version once the data is corrected. Additionally, your assumptions about the `long_manuscript_identifier` are indeed accurate. 

2- The target table for this spreadsheet will also be updated using the same data pipeline, for which we have adjusted the frequency to every 2 hours.

Please let me know if modifying the fields within the spreadsheet would be beneficial.

**Comment by fred-atherden** [comment]
Great to know - thanks very much Hazal. With that understanding, if OK with you, I'm happy with the fields in the spreadsheet as they are.

**Comment by acollings** [comment]
Thanks @HazalCiplak and @fred-atherden I've moved the reviews across to V2.

**Comment by fred-atherden** [comment]
Thanks Andy!
